### DISCUSSION TITLE
canot debug android

### DISCUSSION THREAD

**Discussion by bobwatcherx** [NativeScript/NativeScript#10086] [Q&A]
i have a problem when i want to run android debug.  gradle build runs normally and the debug apk is installed perfectly but I can't save my code and see the changes

command : [bobwatcherx@fedora ~]$  **ns debug android**

log nativescript
```
Project successfully built.
Build time: 87.564 s.
The build result is located at: /home/bobwatcherx/belajar/nativescript/ns1/platforms/android/app/build/outputs/apk/debug/app-debug.apk
Installing on device 192.168.100.2:5555...
Successfully installed on device with identifier '192.168.100.2:5555'.
Unable to apply changes on device: 192.168.100.2:5555. Error is: Invalid Version: null.
```

my Nativescript Version = 8.3.3
node version = v16.18.1

Adb version
[root@fedora ns1]# adb --version
Android Debug Bridge version 1.0.41
Version 31.0.2-android-tools
Installed as /usr/bin/adb

**and the next question**
 is why in the mobile version of nativescript preview when there is an error in coding, nativescript preview doesn't want to run.  and have to scan the QRcode again? it makes me feel a little stuck.  because I have to re-scan the Qrcode again.

**Comment by rigor789** [comment]
Post your `package.json`, feel like it's related to the version of `@nativescript/android` in there.

> is why in the mobile version of nativescript preview when there is an error in coding, nativescript preview doesn't want to run

We have plans to improve the android experience and improve the error handling 👍 iOS should be better at the moment, as most errors can be caught and shown on screen without disconnecting the preview session, and we want to do the same for android in the near future.
